packages/stargate/src/signingstargateclient.ts

```diff
@@ -0,0 +1,188 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { fromBase64 } from "@cosmjs/encoding";
+import {
+  AccountData,
+  buildFeeTable,
+  Coin,
+  CosmosFeeTable,
+  encodeSecp256k1Pubkey,
+  GasLimits,
+  GasPrice,
+  makeSignDoc as makeSignDocAmino,
+  Msg,
+  StdFee,
+  StdSignDoc,
+} from "@cosmjs/launchpad";
+import { Int53 } from "@cosmjs/math";
+import {
+  EncodeObject,
+  encodePubkey,
+  isOfflineDirectSigner,
+  makeAuthInfoBytes,
+  makeSignDoc,
+  OfflineSigner,
+  Registry,
+} from "@cosmjs/proto-signing";
+import { Client as TendermintClient } from "@cosmjs/tendermint-rpc";
+
+import { cosmos } from "./codec";
+import { BroadcastTxResponse, StargateClient } from "./stargateclient";
+
+const { TxRaw } = cosmos.tx.v1beta1;
+
+function snakifyMsgValue(obj: Msg): Msg {
+  return {
+    ...obj,
+    value: Object.entries(obj.value).reduce(
+      (snakified, [key, value]) => ({
+        ...snakified,
+        [key
+          .split(/(?=[A-Z])/)
+          .join("_")
+          .toLowerCase()]: value,
```

Comment:
Can we extract the camel -> snake case conversion in a function that comes with unit tests? There will be some nasty edge cases and then we have a better time debugging it.